openshift-maven-plugin/plugin/src/main/java/org/eclipse/jkube/maven/plugin/mojo/build/OpenshiftResourceMojo.java

```diff
@@ -28,6 +30,48 @@
 @Mojo(name = "resource", defaultPhase = LifecyclePhase.PROCESS_RESOURCES, requiresDependencyResolution = ResolutionScope.COMPILE)
 public class OpenshiftResourceMojo extends ResourceMojo {
 
+    /**
+     * The OpenShift deploy timeout in seconds:
+     * See this issue for background of why for end users on slow wifi on their laptops
+     * DeploymentConfigs usually barf: https://github.com/openshift/origin/issues/10531
+     *
+     * Please follow also the discussion at
+     * <ul>
+     *     <li>https://github.com/fabric8io/fabric8-maven-plugin/pull/944#discussion_r116962969</li>
+     *     <li>https://github.com/fabric8io/fabric8-maven-plugin/pull/794</li>
+     * </ul>
+     * and the references within it for the reason of this ridiculous long default timeout
+     * (in short: Its because Docker image download times are added to the deployment time, making
+     * the default of 10 minutes quite unusable if multiple images are included in the deployment).
+     */
+    @Parameter(property = "jkube.openshift.deployTimeoutSeconds", defaultValue = "3600")
+    private Long openshiftDeployTimeoutSeconds;
+
+    /**
+     * If set to true it would set the container image reference to "", this is done to handle weird
+     * behavior of OpenShift 3.7 in which subsequent rollouts lead to ImagePullErr
+     *
+     * Please see discussion at
+     * <ul>
+     *     <li>https://github.com/openshift/origin/issues/18406</li>
+     *     <li>https://github.com/fabric8io/fabric8-maven-plugin/issues/1130</li>
+     * </ul>
+     */
+    @Parameter(property = "jkube.openshift.trimImageInContainerSpec", defaultValue = "false")
+    private Boolean trimImageInContainerSpec;
+
+    @Parameter(property = "jkube.openshift.generateRoute", defaultValue = "true")
+    private Boolean generateRoute;
+
+    @Parameter(property = "jkube.openshift.enableAutomaticTrigger", defaultValue = "true")
+    private Boolean enableAutomaticTrigger;
+
+    @Parameter(property = "jkube.openshift.imageChangeTrigger", defaultValue = "true")
+    private Boolean enableImageChangeTrigger;
+
+    @Parameter(property = "jkube.openshift.enrichAllWithImageChangeTrigger", defaultValue = "false")
+    private Boolean erichAllWithImageChangeTrigger;
```

Comment:
Are these properties dead? Could you open an issue providing details (if you know about them)?
It seems that at least some of them are used on Enrichers, but are not read from the Mojo.

Reply:
Umm, you mean add javadocs. I see these are covered in plugin documentation: https://www.eclipse.org/jkube/docs/openshift-maven-plugin#Supported-Properties-Resource

Reply:
Ok, let me rephrase. These properties are dead in the Mojo, they are directly computed/retrieved in the Enricher, meaning these fields can be deleted (fields are ignored, same as provided defaults).
So what we need is an issue providing details about what these properties do (link to docs is good), and what should we do with them. Basically, we need an answer to why are they in the Mojo if they are ignored > this seems like the result of an abandoned work in progress. And what we want, complete their movement to the Mojo (properties no longer retrieved from Enricher) **or** complete their movement to the Enricher (delete them from the Mojo).

Reply:
Thanks for explanation. I've filed https://github.com/eclipse/jkube/issues/259 for this